File: completable-reactor-example/src/main/java/ru/fix/completable/reactor/example/SubscribePayload.java
package ru.fix.completable.reactor.example;

import lombok.Data;
import lombok.experimental.Accessors;
import ru.fix.completable.reactor.example.services.ServiceInfo;
import ru.fix.completable.reactor.example.services.UserProfile;
import ru.fix.completable.reactor.example.services.AccountInfo;

/**
 * Created by swarmshine on 16.10.2016.
 */
public class SubscribePayload {

    @Data
    @Accessors()
    public static class Request {
        Long userId;
        Long serviceId;
    }

    @Data
    @Accessors()
    public static class Response {
        Enum status;
        Integer moneyWithdrawed;
    }

    @Data
    @Accessors()
    public static class IntermediateData{
        ServiceInfo serviceInfo;
        AccountInfo accountInfo;
        UserProfile userInfo;
    }

    public final PurchasePayload.Request request = new PurchasePayload.Request();
    public final PurchasePayload.IntermediateData intermediateData = new PurchasePayload.IntermediateData();
    public final PurchasePayload.Response response = new PurchasePayload.Response();

}

File: completable-reactor-example/src/main/kotlin/ru/fix/completable/reactor/example/SubscribePayload.kt
package ru.fix.completable.reactor.example;

import ru.fix.completable.reactor.example.services.ServiceInfo;
import ru.fix.completable.reactor.example.services.UserProfile;
import ru.fix.completable.reactor.example.services.AccountInfo;
import java.math.BigDecimal

/**
 * Created by swarmshine on 16.10.2016.
 */
data class SubscribePayload(val request: Request) {

    val intermediateData = IntermediateData()
    val response = Response()

    data class Request(
            val userId: Long,
            val serviceId: Long
    )

    data class Response(
            var status: Enum<*>? = null,
            var moneyWithdrawed: Int? = null,
            var newAmount: BigDecimal? = null,
            //TODO replace Enum with internal status enum
            var bonusPurchaseStatus: Enum<*>? = null
    )

    class IntermediateData {
        lateinit var serviceInfo: ServiceInfo
        lateinit var accountInfo: AccountInfo
        lateinit var userInfo: UserProfile
    }
}
